crates/agent/src/discovers.rs

```diff
@@ -335,11 +338,28 @@ impl DiscoverHandler {
 
         // Deeply merge the capture and its bindings.
         let capture_name = models::Capture::new(capture_name);
+        let existing_capture = catalog.captures.remove(&capture_name);
+        // This is a hack to prevent autoDiscovers from overwriting changes to the capture endpoint.
```

Comment:
Why is it a hack? What's the non-hacky way to do this?

Reply:
The non-hacky way is to re-implement auto-discovers in controllers. The hack is because that'll take a while

Reply:
Ahhhh, okay that makes sense: auto-discovers aren't controllers yet, that's why this is a problem in the first place. Otherwise this would be taken care of by the controllers' new transaction-less updates mechanism